### PR TITLE
docs: adjust Notary request submission documentation

### PR DESCRIPTION
**Pull request by AnnaShaleva**
All Notary contract witnesses in incomplete transaction (both main and fallback) should have invocation scripts pushing dummy signature on stack. See the 7-th step of the notary request creation documentation: https://github.com/nspcc-dev/neo-go/blob/master/docs/notary.md#2-request-submission.

This bug is closely related to #2955 and prevents the neofs-node code from migration to Notary Actor, as far as the behaviour of notary request creation has been changed.

### CODE REVIEW

File: pkg/rpcclient/notary/actor.go [hunk 1]
```diff
@@ -282,7 +284,10 @@ func (a *Actor) SendRequest(mainTx *transaction.Transaction, fbTx *transaction.T
 	if err != nil {
 		return mainHash, fbHash, vub, err
 	}
-	fbTx.Scripts[0].InvocationScript = append([]byte{byte(opcode.PUSHDATA1), keys.SignatureLen}, make([]byte, keys.SignatureLen)...) // Must be present.
+	// According to the Actor's constructor, Notary contract is always the last sender
+	// of the main transaction and always the first sender of the fallback transaction.
```
Comment (roman-khimov) [inline pkg/rpcclient/notary/actor.go:288]: You can't rely on that. `mainTx` can be anything.

**Comment by roman-khimov** [comment]
But what's the problem with the old code, really?

File: pkg/rpcclient/notary/actor.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/rpcclient/notary/actor.go:288]: Notary subsystem itself doesn't rely on that and is able to work with both empty and dummy witnesses. However, documentation requires proper main transaction to have dummy notary witness. This requirement was added for proper contract-based witness verification cost calculation days ago when Notary fees where calculated on the client side without requests to RPC server. Now this requirement is obsolete and can be removed, I agree.

However, we have user's code relying on this dummy witness. We can safely update the neofs-node code to handle the new case of empty Notary invocation script, then OK, I'll just fix the neo-go's documentation.

Comment (AnnaShaleva) [reply pkg/rpcclient/notary/actor.go]: OK, I've fixed the doc.

Currently we have the following situation: empty Notary witness is allowed for main transaction and prohibited for fallback transaction. I don't quite like this situation, it leads to misunderstanding.

**Comment by AnnaShaleva** [comment]
> But what's the problem with the old code, really?

The old code is OK. The only thing I don't like wrt the current Notary subsystem is that main transactions are allowed to have completely empty Notary witness and fallbacks are obliged to have dummy Notary witness. It's kind of inconsistency.

**Comment by roman-khimov** [comment]
> It's kind of inconsistency.

Can it be fixed?

**Comment by roman-khimov** [comment]
> related to #2995

#2955, btw.

**Comment by AnnaShaleva** [comment]
> Can it be fixed?

Currently we have the requirement of dummy Notary witness specified as a validation rule of `P2PNotaryRequest` payload: https://github.com/nspcc-dev/neo-go/blob/7b109586ca75e50cdd56a76a68955a7506e9eda3/pkg/network/payload/notary_request.go#L112
It is done to be able to properly verify incomplete fallback transaction while pooling it to the mempool and to check that specified fallback's NetworkFee is enough to pay for everything and to satisfy the Notary service needs:
https://github.com/nspcc-dev/neo-go/blob/7b109586ca75e50cdd56a76a68955a7506e9eda3/pkg/core/blockchain.go#L2895-L2900
Thus, we can't remove this fallback's dummy Notary witness requirement.

What we can is to add dummy witness requirement to main transaction (and, in fact, require all Notary signers to have dummy witness for all transactions). But I'm not sure whether we really need it because what the notary service does is ignoring any contract witness during witness check:
https://github.com/nspcc-dev/neo-go/blob/7b109586ca75e50cdd56a76a68955a7506e9eda3/pkg/services/notary/notary.go#L509-L517
and completely replacing Notary witness with the self constructed one while finalizing the transaction:
https://github.com/nspcc-dev/neo-go/blob/7b109586ca75e50cdd56a76a68955a7506e9eda3/pkg/services/notary/notary.go#L399-L408

To sum up, currently the Notary contract witness of fallback transaction is a really special thing, because for other contract witnesses we have the following requirement:
https://github.com/nspcc-dev/neo-go/blob/7b109586ca75e50cdd56a76a68955a7506e9eda3/docs/notary.md?plain=1#L363-L366
This requirement goes according to the code-level implementation and valid for all contract witnesses except the Notary contract witness of the fallback. At the same time I'm not sure we really want to do something with it.

**Comment by roman-khimov** [comment]
> check that specified fallback's NetworkFee is enough to pay for everything

Right.

> we can't remove this fallback's dummy Notary witness requirement.

I'm not so sure about this one. `calculatenetworkfee` handles this easily and so a transaction checking code can. But this may add some overhead.

Overall, we're OK as is. Maybe there'll be a better time to think about it again.

**Review by roman-khimov — APPROVED** [review]
